examples/consumer_rebalance_example/consumer_rebalance_example.go

```diff
@@ -0,0 +1,302 @@
+/**
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Example high-level Apache Kafka consumer demonstrating use of rebalance
+// callback along with manual commit.
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+)
+
+func main() {
+	if len(os.Args) < 5 {
+		fmt.Fprintf(os.Stderr, "Usage: %s <bootstrap-servers> <group> "+
+			"<protocol=EAGER|COOPERATIVE - same for all consumers> <topics..>\n",
+			os.Args[0])
+		os.Exit(1)
+	}
+
+	var err error
+	bootstrapServers := os.Args[1]
+	group := os.Args[2]
+	protocol := strings.ToUpper(os.Args[3])
+	topics := os.Args[4:]
+
+	var strategy string
+	switch protocol {
+	case "EAGER":
+		strategy = "range"
+	case "COOPERATIVE":
+		strategy = "cooperative-sticky"
+	default:
+		fmt.Fprintf(os.Stderr,
+			"protocol should be EAGER or COOPERATIVE, was %s\n", protocol)
+		os.Exit(1)
+	}
+
+	sigchan := make(chan os.Signal, 1)
+	signal.Notify(sigchan, os.Interrupt)
+
+	c, err := kafka.NewConsumer(&kafka.ConfigMap{
+		"bootstrap.servers": bootstrapServers,
+		// Consumer group ID
+		"group.id": group,
+		// Start reading from the first message of each assigned
```

Comment:
About this -- I think we should retain the duplicate, because the person reading the example is likely just starting out, and it's better if the relevant information is in one place rather than them having to jump around files